resources/api/resource.py

```diff
@@ -183,14 +183,12 @@ class ResourceSerializer(ExtraDataMixin, TranslatedModelSerializer, munigeo_api.
 
     def get_can_only_be_reserved_externally(self, obj):
         today = datetime.datetime.today()
-        opening_hours_period_exists = obj.periods.exists()
         resource_has_external_reservation_link = obj.external_reservation_url
-        resource_period_is_not_valid = not obj.periods.filter(end__gte=today).exists()
-        if (
-            not opening_hours_period_exists or
-            resource_has_external_reservation_link or
-            resource_period_is_not_valid
-        ):
+        resource_period_is_valid = obj.periods.exists() and obj.periods.filter(end__gte=today).exists()
+        unit_period_is_valid = obj.unit.periods.exists() and obj.unit.periods.filter(end__gte=today).exists()
+        period_is_invalid = not any([resource_period_is_valid, unit_period_is_valid])
```

Comment:
Instead of doing the queries like that above. I would just put each query inside of this any array. Also think of the order in which you put the queries as the the later queries will only get executed it the previous query is `False`

Reply:
I think it would make it harder to read the code if we put query inside the any array, as they are pretty long, also it seems to be more readable if we assign to variables. That was my reasoning.  Would you still want me to change it ?

Reply:
@wiltso Happy with Deepak's reasoning or do you want changes still?